File: code/ch07/7.1.1_haming_weight.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int num = 0;
        for (int i = 0; i < 32; i++){
            if (n & (1 << i))
                num++;                        
        }
        
        return num;
    }
};

File: code/ch07/7.1.1_haming_weight.java
public class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
        int retVal = 0;
        int mask = 1;
        for (int i = 0; i < 32; i++) {
            if ((n & mask) != 0) {
                retVal++;
            }
            mask <<= 1;
        }
        return retVal;
    }
}